src/gsy_e/models/state.py

```diff
@@ -400,10 +402,6 @@ def total_energy_demanded_Wh(self) -> float:
         """Return the total energy demanded in Wh."""
         return self._total_energy_demanded_Wh
 
-    def get_desired_energy(self, time_slot: DateTime) -> float:
-        """Return the desired energy (based on profile data)."""
-        return self._desired_energy_Wh[time_slot]
```

Comment:
This was just a duplication of `get_desired_energy_Wh()`, so I removed and replaced it.